www_src/lib/dispatcher.js

```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  fire: function (eventName, eventData) {
+    var customEvent = new CustomEvent(eventName, {
+      detail: eventData
+    });
+
+    window.dispatchEvent(customEvent);
+  },
+  on: function (eventName, callback) {
+    window.addEventListener(eventName, function(e) {
+      callback.call(this, e.detail);
```

Comment:
does it matter that the context is now the dispatcher instead of the window? They're both going to be something not related to the thing the event was fired by?